static/js/draw_charts.tsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Component } from "react";
+import { randDomId } from "./util";
+import { fetchStatsData, StatsData } from "./data_fetcher";
+import { drawGroupLineChart, computePlotParams } from "./chart/draw";
+
+const MAX_CHART_WIDTH = 1000;
+const MAX_CHART_HEIGHT = 500;
+
+interface StatVarChipPropsType {
+  statVar: string;
+  color: string;
+  deleteStatVarChip: (statVar: string) => void;
+}
+
+class StatVarChip extends Component<StatVarChipPropsType, {}> {
+  render() {
+    return (
+      <div
+        className="pv-chip mdl-chip--deletable"
+        style={{ backgroundColor: this.props.color }}
+      >
+        <span className="mdl-chip__text">{this.props.statVar}</span>
+        <button className="mdl-chip__action">
+          <i
+            className="material-icons"
+            onClick={() => this.props.deleteStatVarChip(this.props.statVar)}
+          >
+            cancel
+          </i>
+        </button>
+      </div>
+    );
+  }
+}
+
+interface ChartRegionPropsType {
+  chartElem: string;
+  placeIds: string[];
+  statVarsAndMeasuredProps: any[];
+  perCapita: boolean;
+}
+
+class ChartRegion extends Component<ChartRegionPropsType, {}> {
+  measuredPropGroup: { string?: string[] };
+  measuredProps: string[];
+  chartIds: string[];
+  width: number;
+  height: number;
+
+  constructor(props) {
+    super(props);
+
+    // get mprop-statVar dict: { mprop: [statVar1, statVar2, ...]}
+    this.measuredPropGroup = {};
+    this.measuredProps = [];
```

Comment:
no need to have this.measuredProps = [] since all them are the keys of this.measuredPropGroup ?

Reply:
Done